### PR TITLE
[4.1] RavenDB-11326 go over client public API and check if we are not missing cancellation tokens

### PR DESCRIPTION
**Pull request by ppekrol**

### CODE REVIEW

File: src/Raven.Client/Http/RequestExecutor.cs [hunk 1]
```diff
@@ -1077,7 +1075,7 @@ private static Task HandleConflict(JsonOperationContext context, HttpResponseMes
 
             OnFailedRequest(url, e);
 
-            await ExecuteAsync(currentNode, currentIndex, context, command, sessionInfo: sessionInfo).ConfigureAwait(false);
+            await ExecuteAsync(currentNode, currentIndex, context, command, shouldRetry: true, sessionInfo: sessionInfo, token: token).ConfigureAwait(false);
```
Comment (arekpalinski) [inline src/Raven.Client/Http/RequestExecutor.cs]: minor: maybe a bit safer would be to use `shouldRetry: default`

File: src/Raven.Server/Documents/Handlers/BatchRequestParser.cs [hunk 2]
```diff
@@ -741,7 +741,7 @@ private static void ThrowUnexpectedToken(JsonParserToken jsonParserToken, JsonPa
         {
             // Although we using here WithCancellation and passing the token,
             // the stream will stay open even after the cancellation until the entire server will be disposed.
-            var read = await stream.ReadAsync(buffer.Buffer.Array, buffer.Buffer.Offset, buffer.Buffer.Count, token).WithCancellation(token);
+            var read = await stream.ReadAsync(buffer.Buffer.Array, buffer.Buffer.Offset, buffer.Buffer.Count, token);
```
Comment (arekpalinski) [inline src/Raven.Server/Documents/Handlers/BatchRequestParser.cs]: the above comment is no longer valid - we don't use WithCancellation anymore. why did we need to add it here as we pass `token` anyway

File: src/Raven.Client/Documents/DocumentStore.cs [hunk 3]
```diff
@@ -405,10 +406,10 @@ public override CountersOperationExecutor Counters
             }
         }
 
-        public override BulkInsertOperation BulkInsert(string database = null)
+        public override BulkInsertOperation BulkInsert(string database = null, CancellationToken token = default)
```
Comment (ayende) [inline src/Raven.Client/Documents/DocumentStore.cs:409]: This is a breaking binary change, so only for 4.1, maintaining source compatability, right?

Comment (ppekrol) [reply src/Raven.Client/Documents/DocumentStore.cs:409]: Yes, I planned not do do a 4.0 PR at all for this issue.

**Pull request merged**